person/src/com/person/core/model/BasePerson.java

```diff
@@ -0,0 +1,173 @@
+package com.person.core.model;
+
+import com.person.core.exceptions.ValidationException;
+import com.person.core.service.PersonAction;
+import com.person.utils.Validator;
+
+public abstract class BasePerson implements PersonAction {
+    protected String name;
+    protected String lastName;
+    protected String surname;
+    protected String nickname;
+    protected int age;
+    protected String email;
+    protected String  gender;
+
+    private static final String NAME_REGEXP = "^[A-Z][a-z]*$";
+    private static final String NICKNAME_REGEXP = "^[a-z0-9]+$";
+    private static final String MAIL_REGEXP = "^[A-Za-z0-9+_.-]+@[A-Za-z0-9.-]+$";
+
+    protected PersonAction actions;
+
+    public BasePerson() {}
+
+    /**
+     * Set the value of name
+     * @param name the new value of name
+     */
+    public void setName(String name)
+    {
+        if(!Validator.names(name, NAME_REGEXP))
+            throw new ValidationException("Invalid name!");
+
+        this.name = name;
+    }
+
+    /**
+     * Get the value of name
+     * @return the value of name
+     */
+    public String getName()
+    {
+        return name;
+    }
+
+    /**
+     * Set the value of lastName
+     * @param lastName the new value of lastName
+     */
+    public void setLastName(String lastName)
+    {
+        if(!Validator.names(lastName, NAME_REGEXP))
+            throw new ValidationException("Invalid last name!");
+
+        this.lastName = lastName;
+    }
+
+    /**
+     * Get the value of lastName
+     * @return the value of lastName
+     */
+    public String getLastName()
+    {
+        return lastName;
+    }
+
+    /**
+     * Set the value of surname
+     * @param surname the new value of surname
+     */
+    public void setSurname(String surname)
+    {
+        if(!Validator.names(surname, NAME_REGEXP))
```

Comment:
@tatevikb I think it's not good to throw exception in setters, maybe you need to move them to service ?

Reply:
@hhov88 I designed my setters following way: method has to check if the value is acceptable then assign the  value. I have no idea in what part of the service I can move them to. If you'll explain in one example how to move them to service, I'll move them.